### PR TITLE
feat(hogql): Add `hogql` JS template tag for HogQL formatting

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Based on https://github.com/PostHog/posthog/pull/16348#discussion_r1250715125, I added template tag `hogql` for hassle-free formatting of HogQL queries inside the frontend:

```typescript
const column = hogql.identifier('person_id')
const fooValue = 2
const barValue = 'xyz'
const from = dayjs('2023-05-04T02:00:00Z')

hogql`
SELECT ${column} FROM events
    WHERE
        properties.foo = ${fooValue}
        AND properties.bar = ${barValue}
        AND timestamp > ${from}`
=== `
SELECT person_id FROM events
    WHERE
        properties.foo = 2
        AND properties.bar = 'xyz'
        AND timestamp > '2023-05-04 02:00:00' // Project timezone taken into account
```

(the diff is inflated a lot by a refactor of `KEY_MAPPING` into its own file that I had to do to avoid some nasty imports…)

## How did you test this code?

Added unit tests.

### CODE REVIEW

File: frontend/src/queries/utils.ts [hunk 1]
```diff
@@ -272,3 +274,48 @@ export function isHogQlAggregation(hogQl: string): boolean {
         hogQl.includes('max(')
     )
 }
+
+export interface HogQLIdentifier {
+    __hogql_identifier: true
+    identifier: string
+}
+
+function hogQlIdentifier(identifier: string): HogQLIdentifier {
```
Comment (mariusandra) [inline frontend/src/queries/utils.ts:283]: Should we stick to the "uppercase acronyms" convention also in camelcase?

```suggestion
function hogQLIdentifier(identifier: string): HogQLIdentifier {
```

**Review by mariusandra — APPROVED** [review]
Sparks joy!

File: frontend/src/queries/utils.ts (hunk 1 above)
Comment (Twixes) [reply frontend/src/queries/utils.ts:283]: IMO in camel case acronyms should be capitalized like words. E.g. `xmlHttpRequest` is easier to read than `XMLHTTPRequest` (and both would be better than the actual insane `XHLHttpRequest`)

Comment (Twixes) [reply frontend/src/queries/utils.ts:283]: I guess HogQL is kind of a special case, because it's part acronym and part word

**Pull request merged**